diff -x config.sub -x config.guess -x sysdep.h.in -x configure -ur /var/tmp/tonyg/openwrt/pristine-scheme48-1.8/c/unix/socket.c ./c/unix/socket.c
--- /var/tmp/tonyg/openwrt/pristine-scheme48-1.8/c/unix/socket.c	2008-01-14 02:06:41.000000000 -0500
+++ ./c/unix/socket.c	2011-06-14 18:13:39.128597312 -0400
@@ -519,11 +519,11 @@
   RETRY_NULL(host, gethostbyname(machine_name));
   if (host == NULL)
     s48_raise_os_error(h_errno);
-  return s48_enter_byte_vector(host->host_addr, host->host_length);
+  return s48_enter_byte_vector(host->h_addr, host->h_length);
 }
 
 static s48_value
-s48_get_host_by_name_result(void)
+s48_get_host_by_name_result(s48_value event_uid)
 {
   /* won't be used here */
 }
@@ -541,7 +541,7 @@
 }
 
 static s48_value
-s48_get_host_by_address_result(void)
+s48_get_host_by_address_result(s48_value event_uid)
 {
   /* won't be used here */
 }
diff -x config.sub -x config.guess -x sysdep.h.in -x configure -ur /var/tmp/tonyg/openwrt/pristine-scheme48-1.8/configure.in ./configure.in
--- /var/tmp/tonyg/openwrt/pristine-scheme48-1.8/configure.in	2008-02-07 04:49:14.000000000 -0500
+++ ./configure.in	2011-06-14 14:43:18.388600125 -0400
@@ -395,23 +395,13 @@
 
 	  oldCFLAGS="$CFLAGS"
 	  CFLAGS="$CFLAGS $1"
-	  AC_TRY_RUN([#include <pthread.h>
-int
-main(void)
-{
-  pthread_kill(pthread_self(), 0);
-}],
+	  AC_TRY_COMPILE([#include <pthread.h>],[pthread_kill(pthread_self(), 0);],
 	    [flags_result="$1 (compile)"],
 	    [CFLAGS="$oldCFLAGS"])
 
 	  oldLDFLAGS="$LDFLAGS"
 	  LDFLAGS="$LDFLAGS $1"
-	  AC_TRY_RUN([#include <pthread.h>
-int
-main(void)
-{
-  pthread_kill(pthread_self(), 0);
-}],
+	  AC_TRY_COMPILE([#include <pthread.h>],[pthread_kill(pthread_self(), 0);],
 		  [flags_result="$flags_result $1 (link)"
 		   pthreads_done="yes"],
 		  [LDFLAGS="$oldLDFLAGS"])
@@ -427,9 +417,9 @@
        AC_CHECK_HEADERS([pthread.h],[dnl
          pthreads_done="no"
          AC_MSG_CHECKING([Pthreads support])
-	 S48_PTHREAD_TRY_FLAGS([-mt])
 	 S48_PTHREAD_TRY_FLAGS([-pthread])
 	 S48_PTHREAD_TRY_FLAGS([-pthreads])
+	 S48_PTHREAD_TRY_FLAGS([-mt])
 	 S48_PTHREAD_TRY_FLAGS([-mthreads])
 	 S48_PTHREAD_TRY_FLAGS([-thread])
  	])
diff -x config.sub -x config.guess -x sysdep.h.in -x configure -ur /var/tmp/tonyg/openwrt/pristine-scheme48-1.8/Makefile.in ./Makefile.in
--- /var/tmp/tonyg/openwrt/pristine-scheme48-1.8/Makefile.in	2008-01-26 10:06:27.000000000 -0500
+++ ./Makefile.in	2011-06-14 14:46:24.784597252 -0400
@@ -132,6 +132,7 @@
 IMAGE = scheme48.image
 INITIAL = $(srcdir)/build/initial.image
 VM = scheme48vm
+HOST_VM = ./$(VM)
 POSIX_EXTERNAL = c/posix.so
 SRFI_27_EXTERNAL = c/srfi-27.so
 UNIX_OBJS = c/unix/misc.o c/unix/io.o c/unix/fd-io.o c/unix/event.o
@@ -345,7 +346,7 @@
 		scheme/posix/packages.scm scheme/srfi/packages.scm \
 		$(usual-files) build/initial.debug build/build-usual-image
 	$(srcdir)/build/build-usual-image $(srcdir) \
-		"`(cd $(srcdir) && pwd)`/scheme" "`pwd`/c" '$(IMAGE)' './$(VM)' \
+		"`(cd $(srcdir) && pwd)`/scheme" "`pwd`/c" '$(IMAGE)' '$(HOST_VM)' \
 		'$(INITIAL)-$(BIT_SUFFIX)'
 
 ### Fake targets:  all clean install man dist
@@ -361,7 +362,7 @@
 inst-image:
 	rm -f '/tmp/$(IMAGE)'
 	$(srcdir)/build/build-usual-image $(srcdir) "$(SHARE)" "$(LIB)" \
-		"/tmp/$(IMAGE)" "./$(VM)" "$(INITIAL)-$(BIT_SUFFIX)"
+		"/tmp/$(IMAGE)" "$(HOST_VM)" "$(INITIAL)-$(BIT_SUFFIX)"
 	$(INSTALL_DATA) /tmp/$(IMAGE) $(DESTDIR)$(LIB)
 	rm /tmp/$(IMAGE)
 
@@ -409,7 +410,7 @@
 	done
 
 inst-config-script:
-	config_script=$(bindir)/$(CONFIG_SCRIPT) &&			\
+	config_script=$(DESTDIR)$(bindir)/$(CONFIG_SCRIPT) &&		\
 	sed -e 's,@incdir\@,$(incdir),g'				\
 	    -e 's,@libdir\@,$(libdir),g'				\
 	    -e 's,@CC\@,$(CC),g'					\
@@ -491,7 +492,7 @@
 		echo ',open test-suites';				\
 		echo '(define-test-suite all-tests (compiler-tests big-tests tcp-sockets-tests misc-tests cml-tests srfi-tests posix-tests))'; \
 		echo '(run-test-suite all-tests)'			\
-	) | ./$(VM) -i $(IMAGE) -h 8000000
+	) | $(HOST_VM) -i $(IMAGE) -h 8000000
 
 # --------------------
 # Rules from here on down are not essential for the basic installation
@@ -607,7 +608,7 @@
 #	$(MAKE) $(VM) PACKAGES=
 	cd $(srcdir) && \
 	$(RUNNABLE) -a batch <build/filenames.scm
-# or:	./$(VM) -i $(srcdir)/$(INITIAL)-$(BIT_SUFFIX) -a batch <build/filenames.scm
+# or:	$(HOST_VM) -i $(srcdir)/$(INITIAL)-$(BIT_SUFFIX) -a batch <build/filenames.scm
 
 
 # --------------------
@@ -735,7 +736,7 @@
 			   (usual-resumer (lambda (args) \
 					    (command-processor #f args))) \
 			   \"foo\")" \
-	| ./$(VM) -i scheme/debug/mini.image -a batch
+	| $(HOST_VM) -i scheme/debug/mini.image -a batch
 
 
 # --------------------
